packages/tinacms/src/plugins/fields/dateFormat.ts

```diff
@@ -45,13 +45,21 @@ export const parse = (
   _name: string,
   field: DatetimepickerProps
 ) => {
-  const dateFormat =
-    typeof field.dateFormat === 'string'
-      ? field.dateFormat
-      : DEFAULT_DATE_DISPLAY_FORMAT
-  if (typeof val === 'string') {
-    const date = moment(val, dateFormat)
-    return date.isValid() ? date.toDate() : val
+  return format(val, _name, field)
```

Comment:
This method is _actually_ supposed to return a Date, not just the formatted string.

Reply:
`react-datetime` must be okay with using a `string`